Visual Adjustments for Boss Room

jilfranco-unity opened the pull request:
### Description (*)
Visual updates for Boss Room!

Updates include:

- A matte around the border of the map to block the camera from seeing lava, edges of meshes, particles, etc
- Lighting changes and optimizations 
-- Reduced number of realtime lights to 4 (from 27 (oof))
-- Baked 3x1024 and 1x128 lightmaps (about 4MB), these are for area lights to make the lava glow more convincing
- Added little blob shadows to characters (a plane with a transparent soft black circle attached to the physics object in the character prefab)
- Changed the lights for the enemy spawners from point lights to spot lights (still not shadow casting)
- Rotated all of the little pots to be in different directions so they don't look so copy-pasted
- Tightened radii of the torch lights and reduced the amount of realtime lights for each torch from 2 to 1
-- Also added a couple of extra torches
- Upped the contrast and adjusted the ambient lighting of the scene to be a more saturated blue

### Related Pull Requests
Some changes are from notes from [this PR](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop/pull/371)

And now a bunch of before/after screenshots!

Matte to block stuff under level:
![image](https://user-images.githubusercontent.com/89089503/140828054-430d9474-b22a-49c6-8804-2740ee8ca335.png)
![image](https://user-images.githubusercontent.com/89089503/140828079-32f17b08-a64c-4159-92b0-f8c821da9aca.png)


Lava glow:
![image](https://user-images.githubusercontent.com/89089503/140827805-21c1a2f4-957f-4736-a9f9-881f1c674463.png)
![image](https://user-images.githubusercontent.com/89089503/140827855-7153bd1a-6e4a-4b48-97db-df10dfec6ead.png)

Enemy spawners:
![image](https://user-images.githubusercontent.com/89089503/140827975-c868981b-7154-4aa3-be86-62b3ea366ddd.png)

![image](https://user-images.githubusercontent.com/89089503/140827936-c58066d0-ae43-438e-aaa9-7890e17b4db6.png)


Shadows for imps:
![image](https://user-images.githubusercontent.com/89089503/140827713-a174a4a4-8cfe-4ba2-a2a7-3118715735ef.png)
![image](https://user-images.githubusercontent.com/89089503/140827675-3f6090c0-bfa1-47c8-bf64-cb2f730e5300.png)


Contrast and ambient lighting adjustment:
![image](https://user-images.githubusercontent.com/89089503/140827524-51a39649-b016-429a-b951-a5d4912a5f4a.png)
![image](https://user-images.githubusercontent.com/89089503/140827454-ad4b2eac-edd8-4b00-b426-0e40e1923496.png)

jilfranco-unity commented:
Oh I just noticed that in these screen shots the torch flames don't show up--they're still there, but I turned off the simulation of particle systems while I was working because I kept clicking on them :P

chrispope commented:
> Oh I just noticed that in these screen shots the torch flames don't show up--they're still there, but I turned off the simulation of particle systems while I was working because I kept clicking on them by accident :P

Can you add them in and share a screenshot, I would like to see it.

jilfranco-unity commented:
> Can you add them in and share a screenshot, I would like to see it.

Of course! Here's a few:

![image](https://user-images.githubusercontent.com/89089503/140957378-4c1abb1c-896d-4242-bf17-23a0a6c91468.png)
![image](https://user-images.githubusercontent.com/89089503/140957386-a5584276-9421-4603-b973-1ec683c14781.png)
![image](https://user-images.githubusercontent.com/89089503/140957449-b8ef5eaa-40c3-4f94-af97-00cf055df259.png)

fernando-cortez reviewed (commented):
Followed up offline and suggested the following changes:
- Removing shadow casting on matte & removing physics colliders
- Look into repositioning of lava from way below level
- Look into reducing particle count on flame torches & volume max range on audiosource component

Also, StreamingAssets meta folder can be removed.

jilfranco-unity commented:
Updates to this PR

* Removed some unnecessary items from baked global illumination, this brought bake down to 2x1024 maps at about 2.7 MB, and bake time to 10 seconds (from ~25)

* Redid the matte as a single mesh using ProBuilder. It no longer casts shadows, has a collider, or receives any kind of lighting (and is ~250 verts)

* Adjusted overall scene color to be more cyan than purple

* Reduced intensity of the imp door lights 

* Reduction of about 1ms frame time in the area with baked lighting (~6.5 before, ~5.5 after on a dev build)

* Moved lava up in Y a bit to make the lava pools feel fuller

* Turned off some import features on some meshes (like importing lights, importing cameras, that kind of thing)

* Reduced audio radius of the torch flames from 500 (which is basically the whole level) to 10 and capped max particles to 5 or 10, down from 1000

![image](https://user-images.githubusercontent.com/89089503/141213258-a45f880e-30b2-4420-8995-b8a6a18cb75f.png)
![image](https://user-images.githubusercontent.com/89089503/141213265-817391e9-4134-4c35-9853-78430af012f2.png)

chrispope reviewed (dismissed):
Lgtm - very cool update

fernando-cortez reviewed (commented):
![image](https://user-images.githubusercontent.com/75813458/141806792-ff5bc087-307f-4ec8-9821-3ab86239ad50.PNG)

Noticing the missing reference on the new border GameObject. Is this a ProBuilder mesh component? Seems to be working without it though, so I'm curious where the mesh for this object lies.

Nitpicking once more, but if it can be positioned at (0,0,0), then great.

pdeschain commented:
I love how our fire particles now have 5 max particles vs 1000 👍

chrispope commented:
> I love how our fire particles now have 5 max particles vs 1000 👍

We will miss those additional 995 max particles

jilfranco-unity commented:
> ![image](https://user-images.githubusercontent.com/75813458/141806792-ff5bc087-307f-4ec8-9821-3ab86239ad50.PNG)
> 
> Noticing the missing reference on the new border GameObject. Is this a ProBuilder mesh component? Seems to be working without it though, so I'm curious where the mesh for this object lies.
> 
> Nitpicking once more, but if it can be positioned at (0,0,0), then great.

I've moved it to 0,0,0! Also, that missing script reference was the probuilder mesh stuff, so I exported the probuilder mesh and brought it in as a proper mesh asset, so now it just uses that and doesn't need the probuilder mesh stuff :)
